pajbot/bot.py

```diff
@@ -130,21 +136,31 @@ def __init__(self, config: cfg.Config, args: argparse.Namespace) -> None:
         self.app_token_manager = AppAccessTokenManager(self.twitch_id_api, RedisManager.get())
         self.twitch_helix_api: TwitchHelixAPI = TwitchHelixAPI(RedisManager.get(), self.app_token_manager)
 
-        self.bot_user_id = self.twitch_helix_api.get_user_id(self.nickname)
-        if self.bot_user_id is None:
-            raise ValueError("The bot login name you entered under [main] does not exist on twitch.")
+        self.streamer: UserBasics = cfg.load_streamer(config, self.twitch_helix_api)
+        self.channel = f"#{self.streamer.login}"
+
+        self.streamer_display: str = self.streamer.name
+        if "streamer_name" in config["web"]:
+            # Override the streamer display name
+            self.streamer_display = config["web"]["streamer_name"]
 
-        self.broadcaster = self.twitch_helix_api.get_user_basics_by_login(self.streamer)
-        if self.broadcaster is None:
-            raise ValueError("The streamer login name you entered under [main] does not exist on twitch.")
+        self.bot_user: UserBasics = cfg.load_bot(config, self.twitch_helix_api)
 
-        self.streamer_user_id = self.broadcaster.id
+        self.control_hub_user: Optional[UserBasics] = self._load_control_hub(config)
+        self.control_hub_channel: Optional[str]
+        if self.control_hub_user:
+            self.control_hub_channel = f"#{self.control_hub_user.login}"
```

Comment:
shouldn't there be a `else: self.control_hub_channel = None` here since otherwise it'll be unitialized and usages of `bot.control_hub_channel` will result in a `KeyError`?

Regardless of that, might it not be cleaner to go with just `self.control_hub` and ditch `self.control_hub_channel` (to make it consistent with `self.streamer` etc.)

Reply:
You're correcl, I thought `a: Optional[str]` would be default initialized in some way but that's not the case. I've replaced it with `a: Optional[str] = None` instead of the else case, so we don't have to specify the typing in multiple places

I'll keep `self.control_hub_channel` (it's consistent with `self.channel` to not have to do the logic in the two places where whispers take place and in the quit logic